Rubberduck.VBEEditor/Events/AutoCompleteEventArgs.cs

```diff
@@ -1,57 +1,45 @@
 ﻿using System;
+using System.Windows.Forms;
 using Rubberduck.VBEditor.SafeComWrappers.Abstract;
 
 namespace Rubberduck.VBEditor.Events
 {
     public class AutoCompleteEventArgs : EventArgs
     {
-        public AutoCompleteEventArgs(ICodePane pane)
+        public AutoCompleteEventArgs(ICodeModule module, WindowsApi.KeyPressEventArgs e)
         {
-            CodePane = pane;
-            var selection = pane.Selection;
-            using (var module = pane.CodeModule)
+            Character = e.Character;
+            CodeModule = module;
+            CurrentSelection = module.GetQualifiedSelection().Value.Selection;
+            CurrentLine = module.GetLines(CurrentSelection);
+
+            if (e.Key == Keys.Delete || 
+                e.Key == Keys.Back ||
+                e.Key == Keys.Enter)
             {
-                ContentHash = module.ContentHash();
-                var atSelection = module.GetLines(selection);
-                if (string.IsNullOrWhiteSpace(atSelection))
-                {
-                    IsCommitted = true;
-                    OldCode = module.GetLines(selection.PreviousLine);
-                }
-                else
-                {
-                    OldCode = module.GetLines(selection);
-                }
+                Keys = e.Key;
             }
         }
 
         /// <summary>
-        /// The CodePane wrapper for the module being edited.
+        /// <c>true</c> if the character has been handled, i.e. written to the code pane.
+        /// Set to <c>false</c> to swallow the character and prevent the WM message from reaching the code pane.
         /// </summary>
-        public ICodePane CodePane { get; }
+        public bool Handled { get; set; }
 
         /// <summary>
-        /// Indicates whether the line of code held in <see cref="OldCode"/> is committed or not.
+        /// The CodeModule wrapper for the module being edited.
         /// </summary>
-        /// <remarks>
-        /// If the line is committed, <see cref="OldCode"/> is located on the line that precedes the current selection in the <see cref="CodePane"/>.
-        /// </remarks>
-        public bool IsCommitted { get; }
+        public ICodeModule CodeModule { get; }
 
-        /// <summary>
-        /// The content hash for the module before autocompletion. Used to prevent misfiring autocompletes.
-        /// </summary>
-        public string ContentHash { get; }
+        public bool IsCharacter => Keys == default;
+        public char Character { get; }
+        public Keys Keys { get; }
```

Comment:
It might be my unfamiliarity but I'm finding `Character` and `Keys` confusing. Maybe provide a doc comment?

Reply:
Added xml-doc